src/DataGrid.tsx

```diff
@@ -996,8 +996,17 @@ function DataGrid<R, SR, K extends Key>(
 
     const onRowChange = (row: R, commitChanges?: boolean) => {
       if (commitChanges) {
-        updateRow(column, selectedPosition.rowIdx, row);
-        closeEditor();
+        // Fixes two issues when editor is closed by clicking on a different cell
```

Comment:
More forward-looking
```suggestion
        // Prevents two issues when editor is closed by clicking on a different cell
```